lib/OAuth2.php

```diff
@@ -823,7 +819,7 @@ public function grantAccessToken(Request $request = null)
                 $stored = $this->grantAccessTokenRefreshToken($client, $input);
                 break;
             default:
-                if (substr($input["grant_type"], 0, 4) !== 'urn:'
+                if (false === strpos($input['grant_type'], 'urn:')
```

Comment:
that doesn't work, it will allow strings such as `foourn:`.
It should be replaced by `0 !==`